File: csharp/2009/the-imaginarium-of-doctor-parnassus-add.cs
using System;
using System.Threading.Tasks;
using Amazon.DynamoDBv2;
using Amazon.DynamoDBv2.Model;
using System.Collections.Generic;

namespace MovieDatabase
{
    class Program
    {
        static async Task Main(string[] args)
        {
            // Create a MovieRepository instance
            var movies = new MovieRepository();

            // Add "The Imaginarium of Doctor Parnassus" to the database
            // This demonstrates how to insert a new item into DynamoDB
            await movies.InsertAsync(
                title: "The Imaginarium of Doctor Parnassus",
                year: 2009,
                plot: "A traveling theater company gives its audience much more than they were expecting.",
                rating: 6.8
            );

            // Confirm that the movie was added by retrieving it
            var movie = await movies.SelectAsync(
                title: "The Imaginarium of Doctor Parnassus",
                year: 2009
            );

            if (movie != null)
            {
                // The movie was found
                Console.WriteLine($"Movie found: {movie}");
            }
            else
            {
                // The movie was not found
                Console.WriteLine("Movie not found");
            }
        }
    }
}

File: python/2009/the-imaginarium-of-doctor-parnassus-add.py

from movies import MovieRepository

# Create a new movie repository using the default configuration
movies = MovieRepository()

# Add The Imaginarium of Doctor Parnassus to the database
movies.insert(
    title="The Imaginarium of Doctor Parnassus", 
    year=2009, 
    plot="A traveling theater company gives its audience much more than they were expecting.", 
    rating=6.8
)

# Confirm that the movie was added
movie = movies.select(
    title="The Imaginarium of Doctor Parnassus", 
    year=2009
)
if movie:
    # The movie was found
    print(f"Movie found: {movie}")
else:
    # The movie was not found
    print("Movie not found")
    